crates/formality-check/src/adts.rs

```diff
@@ -6,6 +9,20 @@ impl super::Check<'_> {
     pub(super) fn check_adt(&self, adt: &Adt) -> Fallible<()> {
         let Adt { id: _, binder } = adt;
 
+        // names is used to check that there are no name conflicts
+        let mut names = HashSet::new();
+        for Variant { name, fields } in &adt.binder.open().1.variants {
```

Comment:
```suggestion
        for Variant { name, fields } in &adt.binder.skip().variants {
```

Reply:
I saw you merged this change, but `skip` doesn't exist on binder so I changed it to the previously mentioned `peek` method instead.

Reply:
Oh, ooops, thanks!